### DISCUSSION TITLE
Astro DB - Many-to-many relationship

### DISCUSSION THREAD

**Discussion by piotrkulpinski** [withastro/docs#7486] [Ideas]
Hi, 
I'm trying out Astro DB and want to create simple database for my blog. 
I want my posts to have categories, so I'd need a many-to-many relationship between them.

Here's what I have:
```
const Post = defineTable({
  columns: {
    id: column.number({ primaryKey: true, unique: true }),
    name: column.text(),
  },
});

const Category = defineTable({
  columns: {
    id: column.number({ primaryKey: true, unique: true }),
    name: column.text(),
  },
});

const PostCategory = defineTable({
  columns: {
    postId: column.number({ references: () => Post.columns.id }),
    categoryId: column.number({ references: () => Category.columns.id }),
  },
});
```

With such db design, I need to use inner joins with every query I do, right? 
Is there any simpler way to do that? I'd love to be able to define the relations on the main tables and have Astro DB create the relation table for me automatically.

Thanks!

**Comment by ArmandPhilippot** [comment]
We're closing stale and outdated discussion to keep only the discussions still relevant to the current project version and/or the current docs state.

Sorry no one could answer your question before but, since [Astro DB has been removed in Astro 7](https://docs.astro.build/en/guides/upgrade-to/v7/#removed-astrojsdb), I'm going to close this discussion.

If you think this is a mistake and that this discussion should remain open, please feel free to ping me, I'll take a second look at it.

Thank your for your contribution!
